capstone/src/main/java/com/google/sps/servlets/DashboardHandlerServlet.java

```diff
@@ -15,6 +24,20 @@
 @WebServlet("/dashboard-handler")
 public class DashboardHandlerServlet extends HttpServlet {
 
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(new Query(Classroom.CLASSROOM_ENTITY_NAME));
+    
+        List<Classroom> classrooms = new ArrayList<>();
+        for(Entity entity : results.asIterable()){
+            classrooms.add(new Classroom(entity));
+        }
```

Comment:
You can skip/inline all of this with PreparedQuery.asList https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.html#asList-com.google.appengine.api.datastore.FetchOptions- (You can create default fetch options with the FetchOptionsBuilder https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions.Builder.html#withDefaults--)

Reply:
I believe the difference would be that asList("Insert fetch option builer here") would return the classrooms as entities but at least for this implementation we needed the classroom objects (didn't know if you were aware of this/noticed) for the client side of the code.